topic-controller/src/main/java/io/strimzi/controller/topic/Session.java

```diff
@@ -147,80 +149,22 @@ public void start() {
         LOGGER.debug("Starting {}", configMapThread);
         configMapThread.start();
 
-        // Reconcile initially
-        reconcileTopics("initial");
-        // And periodically after that
-        vertx.setPeriodic(this.config.get(Config.FULL_RECONCILIATION_INTERVAL_MS),
-            timerId -> {
-                if (stopped) {
-                    vertx.cancelTimer(timerId);
-                    return;
-                }
-                reconcileTopics("periodic");
-            });
-        LOGGER.info("Started");
-    }
-
-    void reconcileTopics(String reconciliationType) {
-        LOGGER.info("Starting {} reconciliation", reconciliationType);
-        kafka.listTopics(arx -> {
-            if (arx.succeeded()) {
-                Set<String> kafkaTopics = arx.result();
-                LOGGER.debug("Reconciling kafka topics {}", kafkaTopics);
-                // First reconcile the topics in kafka
-                for (String name : kafkaTopics) {
-                    LOGGER.debug("{} reconciliation of topic {}", reconciliationType, name);
-                    TopicName topicName = new TopicName(name);
-                    k8s.getFromName(topicName.asMapName(), ar -> {
-                        ConfigMap cm = ar.result();
-
-                        controller.reconcile(cm, topicName, reconcileResult -> {
-                            if (reconcileResult.succeeded()) {
-                                LOGGER.info("Success {} reconciling ConfigMap {} topic {}",
-                                        reconciliationType, Controller.logConfigMap(cm), topicName);
-                            } else {
-                                LOGGER.error("Error {} reconciling ConfigMap {} topic {}",
-                                        reconciliationType, Controller.logConfigMap(cm), topicName, reconcileResult.cause());
-                            }
-                        });
+        final Long interval = config.get(Config.FULL_RECONCILIATION_INTERVAL_MS);
+        Handler<Long> periodic = new Handler<Long>() {
+            @Override
+            public void handle(Long oldTimerId) {
```

Comment:
what's the reason for having this `oldTimerId` parameter ? It's not used, isn't it ?

Reply:
We're implementing a `Handler<Long>`, so there's necessarily a single `Long` parameter. The `handler` is being passed to `vertx.setTimer()` (line 160), which requires a `Handler<Long>`. So the parameter is the vertx-assigned id of its internal timer which will fire the handler. But the handler reschedules the hander, so there are two timer ids in play here: The one which caused the current invocation of `handle()` and the one returned from `vertx.setTimer()`. I called the first one `oldTimerId` to try to distinguish it from the 'new' one on line 160 (which is a field). Do you have a better name?

Reply:
No suggestions, you are right. I was totally wrong thinking that and `Handler<Void>` would have been enough but I forgot that the `setTimer` needs it. Sorry for the noise ...